runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchPortablePipelineTranslator.java

```diff
@@ -290,6 +296,26 @@ public FlinkPortablePipelineTranslator.Executor translate(
     return context;
   }
 
+  private static <K, V> void translateRedistributeByKey(
+      PTransformNode transform, RunnerApi.Pipeline pipeline, BatchTranslationContext context) {
+    DataSet<WindowedValue<KV<K, V>>> inputDataSet =
+        context.getDataSetOrThrow(
+            Iterables.getOnlyElement(transform.getTransform().getInputsMap().values()));
+    context.addDataSet(
+        Iterables.getOnlyElement(transform.getTransform().getOutputsMap().values()),
+        inputDataSet.rebalance());
```

Comment:
I think for FlinkRunner today, all of Reshuffle, RedistributeArbitrarily, and RedistributeByKey are basically the same semantics. I do think that is a coincidence rather than a logical necessity, which is why I gave them each their own method.

(these are basically just copy/paste of Reshuffle translation)

Reply:
Tried a rewrite. still testing it but I pushed the change up.

Reply:
This can be resolved later.